### PR TITLE
Add xrdcl-record package

### LINKED ISSUE #7826: Add xrootd recorder plugin
There is a request to include xrootd recorder plugin in cmssw env ( https://its.cern.ch/jira/browse/CMSTZ-974 ). @iarspider , can you please include https://github.com/xrootd/xrdcl-record tag v5.4.2 for 12.4.X ?

### PR DESCRIPTION
**Pull request by iarspider**
Closes #7826

### CODE REVIEW

**Comment by iarspider** [comment]
please test

**Comment by smuzaffar** [comment]
@iarspider , we should also add the toolfile and include it in cmssw-tool-conf otherwise bot is not going to build it.

File: xrdcl-record.spec [hunk 1]
```diff
@@ -0,0 +1,23 @@
+### RPM external xrdcl-record 5.4.2
+Source: https://github.com/xrootd/xrdcl-record/archive/refs/tags/v5.4.2.tar.gz
```
Comment (smuzaffar) [inline xrdcl-record.spec]: please use `%{realversion}` instead of hardcoded version string here

**Comment by iarspider** [comment]
please test

File: scram-tools.file/tools/xrdcl-record/xrdcl-record.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<tool name="xrdcl-record" version="@TOOL_VERSION@">
+  <lib name="XrdClRecorder-5"/>
+  <client>
+    <environment name="XRDCL-RECORD_BASE" default="@TOOL_ROOT@"/>
+    <environment name="LIBDIR"            default="$xrdcl-record_BASE/lib64"/>
+  </client>
```
Comment (smuzaffar) [inline scram-tools.file/tools/xrdcl-record/xrdcl-record.xml:4]: As this just provides a plugin for xrootd, so I think we do not need to provide LIBDIR and also remove `<lib name="..."/>` . Instead just add `<runtime name="XRDCL_RECORDER_PLUGIN"       value="${XRDCL_RECORD_BASE}/lib64/libXrdClRecorder-5.so"/>` outside the `<client>` section. this way one someone wants to create the [recorder.conf](https://github.com/xrootd/xrdcl-record) then they can use `$XRDCL_RECORDER_PLUGIN`

File: scram-tools.file/tools/xrdcl-record/xrdcl-record.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<tool name="xrdcl-record" version="@TOOL_VERSION@">
+  <lib name="XrdClRecorder-5"/>
+  <client>
+    <environment name="XRDCL-RECORD_BASE" default="@TOOL_ROOT@"/>
```
Comment (smuzaffar) [inline scram-tools.file/tools/xrdcl-record/xrdcl-record.xml]: please change this to `XRDCL_RECORD_BASE`  i.e use of `_` instead of `-`

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
test parameters:
 - full_bla = true

**Comment by smuzaffar** [comment]
test parameters:
- full_boo = true

**Comment by smuzaffar** [comment]
test parameters:
- full_boo = true

**Comment by smuzaffar** [comment]
test parameters:
- full_boo = true

**Comment by smuzaffar** [comment]
Please test for el8_ppc64le_gcc10

**Comment by smuzaffar** [comment]
Please test for el8_aarch64_gcc10

**Comment by smuzaffar** [comment]
Please test for el9_amd64_gcc11

**Comment by smuzaffar** [comment]
+externals
looks good

**Comment by perrotta** [comment]
> +externals looks good

Great! Thank you @smuzaffar and @iarspider
Is any backport needed to have this also available for 12_3? IOW, since all other ingredients needed to build 12_3_2 have been already merged, what's missing to have also this in 12_3?

**Comment by smuzaffar** [comment]
sure, I will open a PR for 12.3.X now

**Comment by smuzaffar** [comment]
@perrotta , https://github.com/cms-sw/cmsdist/pull/7830 is now available

**Pull request merged**